unlock-app/src/components/interface/checkout/Lock.tsx

```diff
@@ -166,27 +166,54 @@ export const LoadingLock = () => {
 
 interface LockProps {
   lock: RawLock
+  purchasingLockAddress: string | null
+  setPurchasingLockAddress: (lockAddress: string) => void
 }
 
-const lockKeysAvailable = (lock: RawLock) => {
-  if ((lock as any).unlimitedKeys) {
+interface LockKeysAvailableLock {
+  unlimitedKeys?: boolean
+  maxNumberOfKeys?: number
+  outstandingKeys?: number
+}
+
+export const lockKeysAvailable = ({
+  unlimitedKeys,
+  maxNumberOfKeys,
+  outstandingKeys,
+}: LockKeysAvailableLock) => {
+  if (unlimitedKeys) {
     return 'Unlimited'
   }
 
   // maxNumberOfKeys and outstandingKeys are assumed to be defined
   // if they are ever not, a runtime error can occur
-  return (lock.maxNumberOfKeys! - lock.outstandingKeys!).toString()
+  return (maxNumberOfKeys! - outstandingKeys!).toString()
```

Comment:
This may be an artifact of how things were before, but this looks a bit weird. We should probably move the value generation portion of this outside of the view component

Reply:
This actually is outside the component, just contained in the same file at the moment. My next project involves a bit of an overhaul on this piece, and in the process of doing that I'll look for checkout-related things to extract to utils